solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -614,125 +615,77 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
         uint256 id,
         uint32 portion
     ) internal {
-        ProtectedLiquidity memory liquidity = protectedLiquidity(id, provider);
-
-        // verify that the pool is whitelisted
-        _poolWhitelisted(liquidity.poolToken);
-
-        // verify that the protected liquidity is not removed on the same block in which it was added
-        require(liquidity.timestamp < time(), "ERR_TOO_EARLY");
-
-        if (portion == PPM_RESOLUTION) {
-            // notify event subscribers
-            notifyEventSubscribersOnRemovingLiquidity(
-                id,
-                provider,
-                liquidity.poolToken,
-                liquidity.reserveToken,
-                liquidity.poolAmount,
-                liquidity.reserveAmount
-            );
-
-            // remove the protected liquidity from the provider
-            _store.removeProtectedLiquidity(id);
-        } else {
-            // remove a portion of the protected liquidity from the provider
-            uint256 fullPoolAmount = liquidity.poolAmount;
-            uint256 fullReserveAmount = liquidity.reserveAmount;
-            liquidity.poolAmount = liquidity.poolAmount.mul(portion) / PPM_RESOLUTION;
-            liquidity.reserveAmount = liquidity.reserveAmount.mul(portion) / PPM_RESOLUTION;
-
-            // notify event subscribers
-            notifyEventSubscribersOnRemovingLiquidity(
-                id,
-                provider,
-                liquidity.poolToken,
-                liquidity.reserveToken,
-                liquidity.poolAmount,
-                liquidity.reserveAmount
-            );
-
-            _store.updateProtectedLiquidityAmounts(
-                id,
-                fullPoolAmount - liquidity.poolAmount,
-                fullReserveAmount - liquidity.reserveAmount
-            );
-        }
-
-        // update the statistics
-        _stats.decreaseTotalAmounts(
-            liquidity.provider,
-            liquidity.poolToken,
-            liquidity.reserveToken,
-            liquidity.poolAmount,
-            liquidity.reserveAmount
-        );
-
-        // update last liquidity removal checkpoint
-        _lastRemoveCheckpointStore.addCheckpoint(provider);
+        // remove the protected liquidity from the store and update the stats and the last removal checkpoint
+        ProtectedLiquidity memory removedLiquidity = removeProtectedLiquidity(provider, id, portion);
```

Comment:
while you're at it, might as well rename `removedLiquidity` to `removedPosition` etc.

Reply:
Public functions as well or only internally?

Reply:
the new function should be `transferPosition`. add/remove liquidity should of course stay the same.
is there any other public function that has `liquidity` in it?

Reply:
I've finished renaming it (and many various locations), but will open a separate PR for it

Reply:
Such PR technically already exists (#528).

Reply:
We will sync everything it #528, when the time is right.